sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -687,6 +682,15 @@ private Mono<ReadReplicaResult> readPrimaryInternalAsync(
                         true,
                         primaryUriReference.get(),
                         replicaStatusList);
+
+                if (storeTaskException instanceof CosmosException) {
+                    CosmosException cosmosException = (CosmosException) storeTaskException;
+
+                    if (com.azure.cosmos.implementation.Exceptions.isAvoidQuorumSelectionException(cosmosException)) {
```

Comment:
this change also fixes another existing issue I think: 

when document Quorum not selected -> SDK will fallback to primary to read the doc, and without this change, it will keep retrying on the primary for few times